src/fhirtypes/ElementDefinition.ts

```diff
@@ -472,6 +473,56 @@ export class ElementDefinition {
     this._original = undefined;
   }
 
+  clearOriginalProperty(pathParts: PathPart[]): void {
+    let currentOriginalElement: any = this._original;
+    // eslint-disable-next-line
+    let currentActualElement: any = this;
+    const clearPath = this.calculateClearPath(pathParts);
+    for (const [i, pathPart] of clearPath.entries()) {
+      if (i < clearPath.length - 1) {
+        const key = pathPart.primitive ? `_${pathPart.base}` : pathPart.base;
+        currentOriginalElement = currentOriginalElement?.[key];
+        currentActualElement = currentActualElement?.[key];
+        const currentIndex = getArrayIndex(pathPart);
+        if (currentIndex != null) {
+          if (Array.isArray(currentOriginalElement)) {
+            currentOriginalElement = currentOriginalElement[currentIndex];
+          }
+          if (Array.isArray(currentActualElement)) {
+            currentActualElement = currentActualElement[currentIndex];
+          }
+        }
+      } else {
+        if (
+          currentOriginalElement?.[pathPart.base] != null &&
+          currentActualElement?.[pathPart.base] != null
+        ) {
```

Comment:
For this `if` statement, why do we only clear the actual element if the original and it both have non-null values? But we always clear the original element regardless after this block.

Reply:
Okay, I'm not entirely sure why I chose that implementation, but in investigating this, I found a scenario that I missed: `clearPath` points to a primitive array, the parent has nothing at that path, but it _does_ have child elements of that path. I'll update this implementation to handle that scenario.

Reply:
The updated logic makes more sense to me, and it looks like it covers your new test case well. Now I think I have one more question related to this part of the logic. Is there a case when the original element would differ from the actual element? As in, is there a time when `currentOriginalElement?.[pathPart.base]` would not be null but `currentActualElement?.[pathPart.base]` would be null? Maybe I'm just thinking they would never differ because of when we call this function, but that probably isn't something we want the function to depend on.